### DISCUSSION TITLE
remove warnings for unused positional argument

### DISCUSSION THREAD

**Discussion by danlipsa** [microsoft/pyright#3418] [Ideas]
pylint has a way to get rid of warnings for unused positional arguments that you need to have by prefixing them with an underscore _.

https://pycodequ.al/docs/pylint-messages/w0613-unused-argument.html

pyright does not seem to respond to this. Is there a way to disable those warnings? Could it be made to respond to the same _ prefix?

Thanks!
Dan

**Comment by erictraut** [comment]
Pyright never emits warnings or errors for unused parameters.

Pyright does emit "hints" with an "unnecessary" tag if the client requests them. If a client requests such hints, it should display the associated text in a subtle, grayed-out manner. Here's a screen shot from VS Code that shows how this should be displayed.

![Screen Shot 2022-05-02 at 2 24 45 PM](https://user-images.githubusercontent.com/7040122/166330162-4bf24214-e7ad-4a20-b90c-8c5d42d0e2a8.png)

Some clients that request these hints may display them in an incorrect manner, instead treating them like other diagnostics. I suspect that's what you're experiencing. If so, that's a bug in your client or LSP plugin, and you should file an issue with the appropriate project. The client or LSP plugin should either not request hints with "unnecessary" tags (in which case pyright will not provide them) or it should display them correctly.

**Comment by danlipsa** [comment]
I see. Thanks for the clarification.

I am using emacs lsp-mode and this is displayed with green underline - less than a warning which is orange, but still distracting. I found another discussion on this topic:

https://github.com/emacs-lsp/lsp-mode/issues/3104
I'll take this up with that team.
